Add instructions for pairing PS3 controllers (#214)
NUbots/NUbook: src/book/02-system/02-subsystems/06-behaviour.mdx

| SQUARE                    | Toggles locking of the head.               |
| R2                        | Right kick.                                |

> To use this module, build the [PS3Walk role](https://github.com/NUbots/NUbots/blob/main/roles/ps3walk.role) and install it on the robot. Ensure the PS3 controller is paired and connected to the right robot. Then run the `./ps3walk` on the robot to begin controlling it.

Reviewer: Wouldn't this be `./b run ps3walk`?

Reviewer 2: No, not on the real robot. That's only in Docker.

#### Setup

1. Install `bluez`, `bluez-utils`, and `bluez-plugins` if they have not already been installed:
   ```
   pacman -S --needed bluez bluez-utils bluez-plugins
   ```
2. Restart the system:
   ```
   reboot now
   ```
   This is only necessary if the packages were not previously installed.
3. Start the bluetooth service:
   ```
   systemctl start bluetooth.service
   ```

<details>
<summary> Start bluetooth automatically </summary>

Bluetooth can be powered on automatically when the system starts. This enables the controllers to
connect without having to run `systemctl start bluetooth.service` and `bluetoothctl power on`. This

> is not enable by default because we're currently unsure whether wifi and bluetooth are interferring

Reviewer 2: Not sure if you found a different issue, but wheen using the other robot the wifi and bluetooth work fine. The robot you were origionally using had a dodgy connector for bluetooth/wifi. The nuc has been swapped out now for a spare.

Author: Yeah, I wrote this before someone figured that out.

Author: Quick question. What would be an appropriate thing to say here? The bluetooth service isn't enabled by default in the system installation, but we've enabled it on some of the robots. Should bluetooth be turned on for all of them in the install script?

Reviewer 2: Yes, I think it should be set up for all robots when they're initially set up. It would still be good to mention it in this page in case something goes wrong and someone needs to figure out how to do it.

with each other.

1. Run
   ```
   systemctl enable bluetooth.service
   ```
2. Set `AutoEnable=true` in `/etc/bluetooth/main.conf`:
   ```
   [Policy]
   AutoEnable=true
   ```

Now, after restarting the robot, and having previously paired the PS3 controller, the PS3 controller
will connect to the robot automatically.

<Alert type='info'>

> <code>ps3walk</code> still needs to be started in order to control the robot.

Reviewer 2: ```suggestion
`ps3walk` still needs to be started in order to control the robot.
```

Author: I don't think it works inside of an `<Alert>` because it changes into a HTML context.

Reviewer 2: Should work https://nubook.nubots.net/kitchen-sink/alerts

</Alert>

</details>

#### Pairing the PS3 Controller via Bluetooth

1. Unplug and power off the PS3 controller.

> 2. The interactive program `bluetoothctl`, and type the following commands:

Reviewer: This sentence seems to start halfway through

   ```
   power on
   scan on
   discoverable on
   ```

>    If bluetooth is working correrctly, you might see other devices appear and disappear.

Reviewer 2: ```suggestion
   If bluetooth is working correctly, you might see other devices appear and disappear.
```

3. Turn on the PS3 controller by pressing the middle PS button. The four LEDS should start flashing quickly.

> 4. Plug the PS3 controller into the computer.

Reviewer: How come we need to plug it in when pairing via Bluetooth? Is this correct or is there a mixup with the Bluetooth and USB instructions?

Author: There is no mixup. It's an alternative way of pairing a device which is incapable of entering a numeric code. From what I remember, trusting the ps3 controller skips the pairing process and enables it to be connected without being paired.

5. Type `yes` when the following prompt appears:
   ```
   Authorize service 00001124-0000-1000-8000-00805f9b34fb (yes/no)
   ```

The PS3 controller is now paired. You can unplug it and it will remain connected.

<Alert type='warning'>

> The controllers are extremely finicky. Placing them on the table can trigger multiple buttons to be pressed!

Reviewer 2: Maybe also mention shaking it can do this?

</Alert>

<Alert type='info'>

> The steps are based on the Arch Linux <a href="https://wiki.archlinux.org/title/Gamepad#Pairing_via_Bluetooth">guide</a>. If you are having trouble or are connecting a different controller try using their guide as a starting point.

Reviewer 2: ```suggestion
The steps are based on the <a href="https://wiki.archlinux.org/title/Gamepad#Pairing_via_Bluetooth">Arch Linux controller pairing guide</a>. If you are having trouble or are connecting a different controller try using their guide as a starting point.
```
Mostly for clarification and to make the link easier to click.

</Alert>

#### Verifying the PS3 Controller is functioning correctly.

> Verifying the PS3 controller is connected and sending messages is useful for debugging the range, interferance, event codes, and making sure the controller is connect to the right robot.

Reviewer: ```suggestion
Verifying the PS3 controller is connected and sending messages is useful for debugging the range, interference, event codes, and making sure the controller is connect to the right robot.
```

1. Install `joyutils`:
   ```
   pacman -S --needed joyutils
   ```
2. If you are connected via ssh run:
   ```
   jstest --event /dev/input/js0
   ```
   Otherwise, if you're connected to a virtual console:
   ```
   jstest /dev/input/js0
   ```

> 3. Move controller inputs, and press `Cntl-C` to exit.

Reviewer 2: ```suggestion
3. Move controller inputs, and press <kbd>Ctrl</kbd> + <kbd>C</kbd> to exit.
```

| SQUARE                    | Toggles locking of the head.               |
| R2                        | Right kick.                                |

> To use this module, build the [PS3Walk role](https://github.com/NUbots/NUbots/blob/main/roles/ps3walk.role) and install it on the robot. Ensure the PS3 controller is paired and connected to the right robot. Then run the `./ps3walk` on the robot to begin controlling it.

Reviewer 2: For someone not familiar with this, I feel like the different steps required can get confusing. And extra line or two explaining this might be good.

```suggestion
To use this module, build the [PS3Walk role](https://github.com/NUbots/NUbots/blob/main/roles/ps3walk.role) and install it on the robot. Ensure the PS3 controller is paired and connected to the right robot. You can setup bluetooth on the robot and pair the controller using the steps below. Once these have been run once, the robot should automatically connect to the controller when both the robot and controller are powered. When you are ready, run `./ps3walk` on the robot to begin controlling it.
```

> To use this module, build the [PS3Walk role](https://github.com/NUbots/NUbots/blob/main/roles/ps3walk.role) and install it on the robot. Ensure the PS3 controller is paired and connected to the right robot. You can setup bluetooth on the robot and pair the controller using the steps below. Once these have been run once, the robot should automatically connect to the controller when both the robot and controller are powered. When you are ready, run `./ps3walk` on the robot to begin controlling it.

Reviewer 2: ```suggestion
To use this module, build the [PS3Walk role](https://github.com/NUbots/NUbots/blob/main/roles/ps3walk.role) and install it on the robot. Ensure the PS3 controller is paired and connected to the right robot. You can set up bluetooth on the robot and pair the controller using the steps below. Once these have been run once, the robot should automatically connect to the controller when both the robot and controller are powered. When you are ready, run `./ps3walk` on the robot to begin controlling it.
```

#### Setup

1. Install `bluez`, `bluez-utils`, and `bluez-plugins` if they have not already been installed:
   ```
   pacman -S --needed bluez bluez-utils bluez-plugins
   ```
2. Restart the system:
   ```
   reboot now
   ```
   This is only necessary if the packages were not previously installed.
3. Start the bluetooth service:
   ```
   systemctl start bluetooth.service
   ```

<details>
<summary> Start bluetooth automatically </summary>

Bluetooth can be powered on automatically when the system starts. This enables the controllers to
connect without having to run `systemctl start bluetooth.service` and `bluetoothctl power on`. This
is not currently enable by default in the existing installation. Some robots might have this enabled.

1. Run
   ```
   systemctl enable bluetooth.service
   ```
2. Set `AutoEnable=true` in `/etc/bluetooth/main.conf`:
   ```
   [Policy]
   AutoEnable=true
   ```

> Now, after restarting the robot, and having previously paired the PS3 controller, the PS3 controller

Reviewer 2: ```suggestion
Now, after restarting the robot, and having previously paired the PS3 controller (see below), the PS3 controller
```
